datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnectorTask.java

```diff
@@ -264,9 +264,12 @@ protected void sendDatastreamProducerRecord(DatastreamProducerRecord datastreamP
           ((metadata, exception) -> {
             if (exception != null) {
               _logger.warn(
-                  String.format("Detected exception being throw from callback for src partition: %s while sending producer "
-                      + "record: %s, exception: ", srcTopicPartition, datastreamProducerRecord), exception);
-              rewindAndPausePartitionOnException(srcTopicPartition, exception);
+                  String.format("Detected exception being throw from flushless send callback for source "
+                      + "topic-partition: %s with metadata: %s, exception: ", srcTopicPartition, metadata),
+                  exception);
```

Comment:
I think you should move this in a function in AbstractKafkaBasedConnectorTask and use that function and avoid using the map outside the class. You should also set the visibilty to private in that case.

Reply:
Do you want me to move the comment or the synchronized block or both?

Reply:
I initially meant just the synchronized block. Whether to move the warning log as well is your discretion.

Reply:
done, didn't add the comment there.